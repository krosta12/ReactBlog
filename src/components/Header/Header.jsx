import { useState } from 'react';
import { Outlet, useNavigate } from 'react-router-dom';

import Logo from './components/Logo/Logo';

import Button from '../../common/Button/Button';
import { _delete } from '../../API/apiWorker';
import { Texts } from '../../const';

import '../../CSS/styles.css';
import { useDispatch } from 'react-redux';
import { deleteUser } from '../../store/user/actionCreators';

import { logOut, userMeGetter } from '../../API/secondLayer';

function Header(props) {
	const dispatch = useDispatch();

	async function LogOutFunction() {
		logOut(props.isLogin);
		props.isLogin(false);
		localStorage.removeItem('token');

		dispatch(deleteUser());
	}

	props.isLogin(localStorage.getItem('token'));
	const navigate = useNavigate();
	const [name, setName] = useState('');

	if (props.token) {
		userMeGetter().then((el) => {
			setName(el.data.result.name);
		});
	}

	return props.token ? (
		<>
			<div className='Header'>
				<Logo />
				<div className='InnerHeader'>
					<span className='Name'>{name}</span>
					<Button
						text={Texts.logOut}
						onClick={() => {
							LogOutFunction();
						}}
					/>
				</div>
			</div>
			<Outlet />
		</>
	) : (
		<>
			<div className='Header'>
				<Logo />
				<div className='InnerHeader'>
					<Button
						onClick={() => {
							navigate(`/registration`);
						}}
						text='Registration'
					/>
				</div>
			</div>
			<Outlet />
		</>
	);
}

export default Header;
