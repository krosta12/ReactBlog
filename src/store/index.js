const store = {
    user: {
    isAuth: false,//boolean
    name: '',//string
    email: '',//string
    token: '',//string
    },
    courses: [], 
    authors: []
   }