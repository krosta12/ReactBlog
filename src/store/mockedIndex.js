import { configureStore } from '@reduxjs/toolkit';
import mockedCourcesReducer from './courses/mockedActionCreators';
import mokedUserReducer from './user/mockedActionCreators';
import mockedAuthorReducers from './authors/mockedActionCreators';

const RootReducer = {
	mokedUserReducer,
	mockedCourcesReducer,
	mockedAuthorReducers,
};

const store = configureStore({
	reducer: RootReducer,
});

export default store;
