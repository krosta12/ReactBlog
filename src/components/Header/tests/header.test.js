import { Provider } from 'react-redux';
import store from '../../../store/mockedIndex';
import Header from '../Header';
import { findByTestId, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React, { useState } from 'react';

test('nameOfTest', async () => {
	render(
		<Provider store={store}>
			<BrowserRouter>
				<Header isLogin={jest.fn()} token={'jwtToken'} />
			</BrowserRouter>
		</Provider>
	);

	const nameBlock = await screen.findByTestId('nameTest');
	expect(nameBlock).toMatchSnapshot();
});
