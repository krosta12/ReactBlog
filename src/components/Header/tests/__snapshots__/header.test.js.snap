// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nameOfTest 1`] = `
<span
  class="Name"
  data-testid="nameTest"
>
  admin
</span>
`;
